docs/variant.md

```diff
@@ -1,207 +1,200 @@
+will support this syntax. Any recent version of Reason will support inline
+records though._
 
-```reason
-[@bs.module "myLibrary"] external drawFloat : float => unit = "draw";
-[@bs.module "myLibrary"] external drawString : string => unit = "draw";
-```
+## Tips
 
-BuckleScript also provides [a few other ways](https://bucklescript.github.io/docs/en/function.html#modeling-polymorphic-function) to do this.
+### Unbound Constructor Error
 
-### Variant Types Are Found By Field Name
+The variant type needs to be in scope when referencing any constructor of that
+variant. An easy way to fix this error is with an explicit type annotation:
 
-Please refer to this [record section](record.md#record-types-are-found-by-field-name). Variants are the same: a function can't accept an arbitrary constructor shared by two different variants. Again, such feature exists, it's called a polymorphic variant. We'll talk about this in the future =).
+```reason
+module Animal = {
+  type t =
+    | Cat
+    | Dog
+    | Horse
+    | Snake;
+};
+
+/* Error: Unbound constructor */
+let x = Cat;
+
+/* No error */
+let y: Animal.t = Cat;
+```
 
-## Design Decisions
+### Tuples as Arguments
 
-Variants, in their many forms (polymorphic variant, open variant, GADT, etc.), are likely _the_ feature of a type system such as Reason's. The aforementioned `option` variant, for example, obliterates the need for nullable types, a major source of bugs in other languages. Philosophically speaking, a problem is composed of many possible branches/conditions. Mishandling these conditions is the majority of what we call bugs. **A type system doesn't magically eliminate bugs; it points out the unhandled conditions and asks you to cover them**\*. The ability to model "this or that" correctly is crucial.
+There is a subtle difference between constructors that accept two arguments and
+constructors that accept one argument that is a 2-tuple. Pay close attention
+to which kind of variant you are dealing with:
 
-For example, some folks wonder how the type system can safely eliminate badly formatted JSON data from propagating into their program. They don't, not by themselves! But if the parser returns the `option` type `None | Some(actualData)`, then you'd have to handle the `None` case explicitly in later call sites. That's all there is.
+```reason
+type oneArgument =
+  | OneArg((int, string));
 
-Performance-wise, a variant can potentially tremendously speed up your program's logic. Here's a piece of JavaScript:
+type twoArguments =
+  | TwoArgs(int, string);
 
-```js
-let data = 'dog';
-if (data === 'dog') {
-  ...
-} else if (data === 'cat') {
-  ...
-} else if (data === 'bird') {
-  ...
-}
+let x = OneArg((1, "one"));
+let y = TwoArgs(2, "two");
 ```
 
-There's a linear amount of branch checking here (`O(n)`). Compare this to using a Reason variant:
+Using a one-argument variant that accepts a 2-tuple is common when dealing with
```

Comment:
Not sure this is super helpful, as it's not immediately obvious why this is the case (even though it's true). If you're looking for any lines to cut, this could be a candidate.

Reply:
Yeah I just wanted a little bit of motivation for the section, otherwise I feel like people wouldn't understand when this comes up